### PR TITLE
feat: [FFM-5160]: updated exports to be used within ff-ui

### PR DESCRIPTION
**Pull request by crystal-fong**
### Summary
Exports needed to allow `ff-ui` to get environment data
Preemptively including the `useDeleteEnvironment` hook for an upcoming ticket

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by crystal-fong** [comment]
retrigger titlecheck

File: src/modules/75-cf/FFCustomMicroFrontendProps.types.ts [hunk 1]
```diff
@@ -6,23 +6,38 @@
...
 import type SectionNoData from './components/NoData/SectionNoData/SectionNoData'
 import type { EnvironmentSDKKeyType } from './utils/CFUtils'
 
 export interface FFCustomMicroFrontendProps {
-  ffServices: typeof ffServices & { useCDGetEnvironment: typeof useGetEnvironment }
+  ffServices: typeof ffServices & {
+    useCDGetEnvironment: typeof useGetEnvironment
+    useGetEnvironmentListForProject: typeof useGetEnvironmentListForProject
+  }
+  customMutations: {
```
Comment (knagurski) [inline src/modules/75-cf/FFCustomMicroFrontendProps.types.ts:34]: Any reason to call these out specifically as mutations? `ffServices` includes dozens of mutations

File: src/modules/75-cf/FFCustomMicroFrontendProps.types.ts [hunk 2]
```diff
@@ -6,23 +6,38 @@
...
 import type * as trackingConstants from '@common/constants/TrackingConstants'
 import type useActiveEnvironment from './hooks/useActiveEnvironment'
 import type SectionNoData from './components/NoData/SectionNoData/SectionNoData'
 import type { EnvironmentSDKKeyType } from './utils/CFUtils'
 
 export interface FFCustomMicroFrontendProps {
-  ffServices: typeof ffServices & { useCDGetEnvironment: typeof useGetEnvironment }
+  ffServices: typeof ffServices & {
+    useCDGetEnvironment: typeof useGetEnvironment
+    useGetEnvironmentListForProject: typeof useGetEnvironmentListForProject
```
Comment (knagurski) [inline src/modules/75-cf/FFCustomMicroFrontendProps.types.ts:32]: It's worth prefixing these with `CD` to make it super obvious in the MFE that these are coming from outside FF
```suggestion
    useCDGetEnvironmentListForProject: typeof useGetEnvironmentListForProject
```

File: src/modules/75-cf/pages/FFUIApp/FFUIApp.tsx [hunk 3]
```diff
@@ -28,12 +35,20 @@ const FFUIMFEApp = lazy(() => import('ffui/MicroFrontendApp'))
...
   <ChildAppMounter<FFCustomMicroFrontendProps>
     ChildApp={FFUIMFEApp}
-    ffServices={{ ...ffServices, useCDGetEnvironment }}
-    customHooks={{ useConfirmAction, useActiveEnvironment, useSyncedEnvironment }}
+    ffServices={{
+      ...ffServices,
+      useCDGetEnvironment,
+      useGetEnvironmentListForProject
+    }}
+    customMutations={{
```
[the remark this answers is not in the document]
Comment (knagurski) [reply src/modules/75-cf/pages/FFUIApp/FFUIApp.tsx:43]: That's probably coz we have services with he same name, so the types are clashing. Prefix anything we get from other services (like CD) and the types should resolve

File: src/modules/75-cf/FFCustomMicroFrontendProps.types.ts (hunk 2 above)
Comment (crystal-fong) [reply src/modules/75-cf/FFCustomMicroFrontendProps.types.ts:32]: ooh okay, I'll try that now 👍

Comment (crystal-fong) [reply src/modules/75-cf/FFCustomMicroFrontendProps.types.ts]: @knagurski that worked! thank you

File: jest.config.js [hunk 4]
```diff
@@ -48,6 +48,7 @@ module.exports = {
     '!src/modules/70-pipeline/components/PipelineStudio/ExecutionGraph/**',
     '!src/modules/25-governance/**', // 25-governance will be moved to a separate micro-frontend repository shortly
     '!src/modules/75-cd/factory/**',
+    '!src/modules/75-cf/pages/FFUIApp/FFUIApp.tsx',
```
Comment (crystal-fong) [inline jest.config.js:51]: @knagurski is this it?

Comment (knagurski) [reply jest.config.js:51]: It is indeed!
Screenshot 2022-11-15 at 18 01 51

**Review by knagurski — APPROVED** [review]
Noice!

**Review by willd0yle — APPROVED** [review]

**Pull request merged**